src/salesforce_requests_oauthlib/__init__.py

```diff
@@ -122,61 +126,66 @@ def __init__(self, client_id, client_secret, username,
         super(SalesforceOAuth2Session, self).__init__(
             client_id=client_id,
             redirect_uri=self.callback_url,
-            client=LegacyApplicationClient(
+            client=oauth2client if oauth2client else LegacyApplicationClient(
                 client_id=client_id
             ) if password is not None else None
         )
 
-        if settings_path is None:
-            settings_path = default_settings_path
-        self.settings_path = settings_path
-
-        if not os.path.exists(self.settings_path):
-            try:
-                os.makedirs(self.settings_path)
-            except OSError as e: # Guard against race condition
-                if e.errno != errno.EEXIST:
-                    raise e
-
-        self.refresh_token_filename = os.path.join(
-            self.settings_path,
-            default_refresh_token_filename
-        )
+        if isinstance(oauth2client, ServiceApplicationClient):
+            import time
```

Comment:
I prefer this at the top of the file with the other imports.  (I know I break this rule in launch_webbrowser_flow(), but that's a library bug workaround so I let it slide there.)